pinot-core/src/main/java/com/linkedin/pinot/core/startree/v2/AggregationFunctionFactory.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.startree.v2;
+
+import javax.annotation.Nonnull;
+import com.linkedin.pinot.core.query.aggregation.function.AggregationFunctionType;
+
+
+/**
+ * Factory class to create instances of aggregation function of the given name.
+ */
+public class AggregationFunctionFactory {
+
+  public AggregationFunctionFactory() {
```

Comment:
private so that no one can construct a factory instance

Reply:
Done